File: StoreApp.Domain.Entity/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace StoreApp.Domain.Entity
{
    public class Product : Infra.Entity.EntityBase
    {
        public virtual string Name { get; set; }

        public virtual string Description { get; set; }

        public virtual decimal Price { get; set; }

        public virtual ICollection<ProductItem> Items { get; set; }

        public virtual bool IsActive { get; set; }

        //public void AddItem(ProductItem item)
        //{
        //    var itemOfProduct = Items.FirstOrDefault(x => x.Item.Id == item.Item.Id);
        //    if (itemOfProduct != null)
        //    {
        //        itemOfProduct.Count = item.Count;
        //    }
        //    else
        //    {
        //        item.Procduct = this;
        //        Items.Add(item);
        //    }   
        //}

        public Product()
        {
            Items = new List<ProductItem>();
        }
    }
}

File: StoreApp.Web/ClientAngularApp/app/models/product.ts
/*
This is the main module of an Angular App
It contains the declarations of every others (Main or not) modules, or components or services
*/

export class Product
{
    public id: number;
    public name: string;
    public description: string;
    public price: number;
    public isActive: boolean;
    public items: ProductItem[] = new Array<ProductItem>();
}

export class ProductItem {
    public id: number;
    public name: string;
    public count: number = 0;
}
